docs/compute-resources.md

```diff
@@ -52,6 +52,251 @@ requirements for `Step`containers, they must be treated as if they are running i
+
+`Task`-level resources requirements can be configured in `Task.spec.resources` or `PipelineRun.spec.taskRunSpecs.resources`.
+
+e.g.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: foo 
+spec:
+  steps:
+    - name: foo1
+    - name: foo2
+  resources:
+    requests:
+      cpu: 1 
+    limits:
+      cpu: 2
+```
+
+### Applying Resources Requirements
+
+**Requests**
+`Task`(`Pod`) level resources requests will be applied only on one `Step`(container) to acquire the same effective request for the `Task`. (An exception with `LimitRange` will be explained in the below section.)
+
+**Limits**
+Different from requests, `Task` level resources limits will be applied on each `Step` as a non-limits-configured `Step` will have higher limits than configured ones by Kubernetes. Although the summed-up limits will lead to a value higher than the desired for `Task`, the effective limits will not be used for scheduling a `Pod` but applied in container runtime. So during any time of the `Task`, this limit will be enforced on each `Step`.
+
+### Interaction with Step-level Resources Requirements
+
+When Tekton now supports configuring step-level resources requirements, from `Task.spec.step.resources`, `Task.spec.stepTemplate.resources`, `TaskRun.spec.stepOverrides.resources`, and `PipelineRun.spec.taskRunSpecs.stepOverrides.resources`, either the task-level or step-level will be allowed to configure resources requirements, therefore avoiding possible conflicts.
+
+The following configurations will be rejected during the webhook validation:
+
+Case #1: `Task.spec.step.resources` and `Task.spec.resources`
+
+```yaml
+kind: Task
+spec:
+  steps:
+    - name: foo
+      requests:
+        cpu: 1
+  resources:
+    requests:
+      cpu: 2
+```
+
+Case #2: `Task.spec.stepTemplate.resources` and `Task.spec.resources`
+
+```yaml
+kind: Task
+spec:
+  steps:
+    - name: foo
+  stepTemplate:
+    requests:
+      cpu: 1
+  resources:
+    requests:
+      cpu: 2
+```
+
+Case #3: `TaskRun.spec.stepOverrides.resources` and `TaskRun.spec.taskSpec.resources`
+
+```yaml
+kind: TaskRun
+spec:
+  stepOverrides:
+    - name: foo
+      resources:
+        requests:
+          cpu: 1
+  taskSpec:
+    resources:
+      requests:
+        cpu: 2  
+```
+
+Case #4: `PipelineRun.spec.taskRunSpecs.stepOverrides.resources` and `PipelineRun.spec.taskRunSpecs.resources`
+
+```yaml
+kind: PipelineRun
+spec:
+  taskRunSpecs:
+    - pipelineTaskName: foo
+      stepOverrides:
+        - name: foo 
+          resources:
+            requests:
+              cpu: 1 
+      resources:
+        requests:
+          cpu: 2 
+```
+
+### Interaction with Runtime Overwriting
```

Comment:
Suggested wording:

Section Name: Specifying compute resources in TaskRuns

Description: You can specify compute resources at the Step level or at the Task level in TaskRuns, but not both.
To specify compute resources at the Step level, use TaskRuns.StepOverrides. <insert example>
To specify compute resources at the Task level, use TaskRuns.Resources. <insert example>

Reply:
agree this idea.

so even one of the mentioned fields is on `PipelineRun`, still be good for us to mention "in `TaskRun`" right? as the field is under `PipelineRun.TaskRunSpecs`

Reply:
closed - updated as `Specifying compute resources in TaskRun/PipelineRun`